swift/ql/test/library-tests/dataflow/taint/core/simple.swift

```diff
@@ -86,3 +86,45 @@ func taintThroughBitwiseOperators() {
 
   sink(arg: ~source()) // $ tainted=87
 }
+
+// --- globals and class member variables ---
+
+let g1 = source()
+let g2 = source() + 1
+let g3 = g1
+var gv = source()
+
+class MyClass {
+	let m1 = source()
+	let m2 = source() + 1
+	let m3 = g1
+	var mv = source()
+
+	static let s1 = source()
+	static let s2 = source() + 1
+	static let s3 = s1
+	static var sv = source()
+
+	func test() {
```

Comment:
You might want to distinguish the case where `test` has no calls and the case where there's an actual construction on a `MyClass` object in which `test` is subsequently called. Comparing to Java, only the latter case will currently have flow from instance initializers. As-is the test would therefore fail to show you progress when the initializers are reachable from the constructor.

Reply:
Thanks for reviewing - I hadn't realized this might be an issue.